### PR TITLE
Refactored Button Component

### PR DESCRIPTION
**Pull request by Johannafendrich**
Storybook preview:
https://my-homie-refactoring-bu-hcgx94.herokuapp.com/storybook/index.html?path=/story/buttons--login-button

### CODE REVIEW

File: client/src/components/Button.js [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
   text-transform: uppercase;
-  font-size: 14px;
-  font-family: Oswald, sans-serif;
-  font-weight: bold;
+  font: bold 0.8rem 'Oswald', sans-serif;
+
   cursor: pointer;
 
   &:active {
     transform: translateY(5px) scale(1.1);
```
Comment (sascharissling) [inline client/src/components/Button.js:18]: The button looks great but I'd expect it to rather decrease than increase it's size when clicking it. To make it feel like i'm pushing something.
just a UI/UX thought :-)

File: client/src/components/Button.js [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
+
+const ButtonSize = styled.div`
+  display: flex;
+  justify-content: center;
+  > * {
+    flex-basis: 50%;
+  }
+`;
+
+const Button = ({ label, onClick }) => (
```
[the remark this answers is not in the document]
Comment (lmachens) [reply client/src/components/Button.js:31]: This file could look like this if you create another file for the layout/container (see above comment):
```jsx
import styled from '@emotion/styled';

const Button = styled.button`
  background-color: #8645ff;
  color: white;
  padding: 0.7rem;
  border: none;
  border-radius: 20px;
  box-shadow: 0px 10px 15px rgba(0, 0, 0, 16%);
  text-transform: uppercase;
  font: bold 0.8rem 'Oswald', sans-serif;
  cursor: pointer;

  &:active {
    transform: translateY(5px) scale(1.1);
    box-shadow: 0px 10px 15px rgba(134, 69, 255, 60%);
  }
`;

export default Button;
```

The layout could look like this:

```jsx
import styled from '@emotion/styled';

// Not the best name, maybe you have a better idea?
const CenterContainer = styled.div`
  display: flex;
  justify-content: center;

  > * {
    flex-basis: 50%;
  }
`;

export default CenterContainer;
```

And your stories:

```jsx
export const loginButton = () => (
  <CenterContainer>
    <Button onClick={action('link to Login')}>Login</Button>
  </CenterContainer>
);
```

File: client/src/components/Button.js (hunk 1 above)
Comment (Johannafendrich) [reply client/src/components/Button.js:18]: I don't know what I was thinking 😅I agree it doesn't make much sense... Thanks for your advice, I changed it and it's much better this way. 👌🏼

File: client/src/components/Button.js [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
+
   cursor: pointer;
 
   &:active {
     transform: translateY(5px) scale(1.1);
     box-shadow: 0px 10px 15px rgba(134, 69, 255, 60%);
   }
 `;
+
+const ButtonSize = styled.div`
```
[the remark this answers is not in the document]
Comment (Johannafendrich) [reply client/src/components/Button.js]: Thank you for this hint 👌🏼The good thing now is that I can insert the new parents container component also in other places, thank you 😀

File: client/src/components/Button.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Johannafendrich) [reply client/src/components/Button.js]: Thanks for the helpful input. In this case, I wanted to save code in the wrong place and made things more complicated. 😅

File: client/src/components/Button.js [hunk 4]
```diff
@@ -1,21 +1,42 @@
+import React from 'react';
 import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
 
-const Button = styled.button`
+const Buttons = styled.button`
```
[the remark this answers is not in the document]
Comment (Johannafendrich) [reply client/src/components/Button.js]: Absolutely, this naming was confusing 🙈

**Pull request merged**